pkg/virt-handler/vm.go

```diff
@@ -2974,3 +2973,19 @@ func (d *VirtualMachineController) reportTargetTopologyForMigratingVMI(vmi *v1.V
 	vmi.Status.MigrationState.TargetNodeTopology = string(topology)
 	return nil
 }
+
+func (d *VirtualMachineController) handleMigrationAbort(vmi *v1.VirtualMachineInstance, client cmdclient.LauncherClient) error {
+	if vmi.Status.MigrationState.AbortStatus == v1.MigrationAbortInProgress {
+		return nil
+	}
+
+	err := client.CancelVirtualMachineMigration(vmi)
+	if err != nil && err.Error() != migrations.CancelMigrationFailedVmiNotMigratingErr {
+		// If migration did not even start there is no need to cancel it
+		log.Log.Object(vmi).Infof("skipping migration cancellation since vmi is not migrating")
```

Comment:
this log message doesn't line up with the logic.

Reply:
So do you suggest to remove this `if` statement altogether?
I understand your point since the condition should never be true, but I wonder if we want to provide an error on such situations in general. I mean, generally, is canceling a non-running migration wrong? should it result with an error?

Reply:
it's useful to have the log message when cancelling a non-running migration. I just meant that the logic doesn't line up

`err.Error() != migrations.CancelMigrationFailedVmiNotMigratingErr` so if the error isn't "not migrating" then we log a message about the vmi not migrating?! I don't follow the logic

Reply:
Oh :man_facepalming: of course!
Done, thanks for catching that